### PR TITLE
Upgrade spikeinterface tools to `SortingAnalyzer`

### PR DESCRIPTION
**Pull request by alejoe91**
and remove `WaveformExtractor`

Let's keep this as a draft until SI v0.101.0 is released

### CODE REVIEW

**Comment by alejoe91** [comment]
Temporarily pinned SI to 0.101.0rc0, we can switch to >=0.101.0 when released

**Comment by alejoe91** [comment]
Thanks @h-mayorquin 

Yes then I would suggest to merge this first! :)

File: requirements-rtd.txt [hunk 1]
```diff
@@ -5,5 +5,5 @@ readthedocs-sphinx-search==0.1.2
 sphinx-toggleprompt==0.2.0
 sphinx-copybutton==0.5.0
 roiextractors  # Needed for the API documentation
-spikeinterface  # Needed for the API documentation
+spikeinterface==0.101.0  # Needed for the API documentation
```
Comment (h-mayorquin) [inline requirements-rtd.txt:8]: Why pinning instead of lower `>=` ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -110,6 +110,9 @@
 ### Testing
 * Add general test for metadata in-place modification by interfaces. [PR #815](https://github.com/catalystneuro/neuroconv/pull/815)
 
+### Dependencies
+* Updated `spikeinterface` to `v0.101.0` to use `SortingAnalyzer` instead of `WaveformExtractor` in the `spikeinterface` tools
```
Comment (h-mayorquin) [inline CHANGELOG.md:114]: Can you add that we are deprecating waveform extractor?

File: src/neuroconv/tools/spikeinterface/spikeinterface.py [hunk 3]
```diff
@@ -1464,8 +1468,8 @@ def add_waveforms(
     )
 
 
-def write_waveforms(
```
Comment (h-mayorquin) [inline src/neuroconv/tools/spikeinterface/spikeinterface.py:1467]: Can you keep these functions with the previous signature and they  throw and informative error of when was the last version of neuroconv where they were supported if they are called? I don't want to throw an import error without context if someone is relying on this.

**Review by h-mayorquin — CHANGES_REQUESTED** [review]
OK, just let's not add some nice deprecations and add these information into the changelog.

File: requirements-rtd.txt (hunk 1 above)
Comment (alejoe91) [reply requirements-rtd.txt:8]: I followed the other pins, used >= for the main requirements

File: src/neuroconv/tools/spikeinterface/spikeinterface.py [hunk 4]
```diff
@@ -1563,6 +1568,33 @@ def write_waveforms(
         )
 
 
+def write_waveforms(
```
Comment (h-mayorquin) [inline src/neuroconv/tools/spikeinterface/spikeinterface.py:1571]: Cool, two things and this is done:
1) Add one exactly like this for `add_waveforms`
2) We have a six months deprecation in neuroconv. Write in the NotImplementedError that the function will be completley removed at some point after February 2025

Comment (alejoe91) [reply src/neuroconv/tools/spikeinterface/spikeinterface.py:1571]: done in last commit!

**Review by h-mayorquin — APPROVED** [review]
This is ready to go. My modularize later by breaking it into at least the following two blocks:
- Add templates (mean, std)
- Add quality metrics

**Pull request merged**